extension/src/commands.ts

```diff
@@ -0,0 +1,40 @@
+import * as vscode from "vscode";
+import * as nls from 'vscode-nls';
+import { openWebHelp } from './help/openWebHelp';
+import { showErrorMessage } from './project/projectCommands';
+
+const localize = nls.config({ messageFormat: nls.MessageFormat.file })();
+
+export function registerCommands(context: vscode.ExtensionContext){
+
+	// Associated with the right click "Open Online Help" menu item
+	context.subscriptions.push(vscode.commands.registerCommand('autolisp.openWebHelp', async () => {
+		try {
+			await openWebHelp();
+		}
+		catch (err) {
+			if (err){
+				let msg = localize("autolispext.help.commands.openWebHelp", "Failed to load the webHelpAbstraction.json file");
+				showErrorMessage(msg, err);
+			}
+		}
+	}));
+	
+
+	// Associated with the right click "Insert Region" menu item
+	context.subscriptions.push(vscode.commands.registerCommand("autolisp.insertFoldingRegion", async () => {
+		try {
+			let commentChar = vscode.window.activeTextEditor.document.fileName.toUpperCase().slice(-4) === ".DCL" ? "//" : ";";
+			const snip = new vscode.SnippetString(commentChar + "#region ${1:description}\n${TM_SELECTED_TEXT}\n" + commentChar + "#endregion");
```

Comment:
The region can be nested? or what will happen if user create nested region?

Reply:
Regions do nest, I can add a GIF of that if you'd like

Reply:
ok, I only remind us that there is a such case, and I think tester will also cover this case.

Reply:
is it able to trigger this command when there's no active text at all? or what if the active text is not lsp or dcl?

Reply:
Yes, this does trigger with no selection. It would just create an empty start/stop region block.

I believe the "when" property on the _package.json_ defines this for the "autolisp" and "autolispdcl" file selectors. So, if you were to open a *.TXT file, this won't even show up as an option. I tested this on a *._LS" file and the context menu option was missing.

Reply:
> Yes, this does trigger with no selection. It would just create an empty start/stop region block.
> 
> I believe the "when" property on the _package.json_ defines this for the "autolisp" and "autolispdcl" file selectors. So, if you were to open a *.TXT file, this won't even show up as an option. I tested this on a *._LS" file and the context menu option was missing.

The "when" is inside and for a context menu item definition, but as shown in your GIF, as long as the lisp extension has been activated, the command can also be directly executed via. the Command Palette of VS Code.
Please have a test:
1. Open a LSP to activate the extension
2. Open a txt file
3. Open the Command Palette and execute the new added command